### PR TITLE
Add width/height attributes to the picture source

### LINKED ISSUE #2746: Set picture source width/height attributes once supported by browers
We added `width` and `height` attributes to the `` element in #940

It looks like browsers will support these attributes on `<source>` elements too in the future. See https://github.com/whatwg/html/pull/5894 and https://groups.google.com/a/chromium.org/g/blink-dev/c/GEjjJE0LCnM/m/0p8CJ48hAgAJ

As soon as the first browser ships this in a stable version we should add these attributes I think.

### PR DESCRIPTION
**Pull request by ausi**
| Q                | A
| -----------------| ---
| Fixed issues     | Fixes #2746

We should probably wait for Chrome 90 to be released on April 13th before merging this. https://chromestatus.com/feature/5737185317748736

UPDATE: Implementation tickets for Safari and Firefox:

- <https://bugzilla.mozilla.org/show_bug.cgi?id=1694741>
- <https://bugs.webkit.org/show_bug.cgi?id=222368>

### CODE REVIEW

**Comment by m-vo** [comment]
This will also need to be adjusted in `Image/Studio/_macros.html.twig` for Contao 4.11.

File: core-bundle/src/Resources/contao/templates/picture/picture_default.html5 [hunk 1]
```diff
@@ -2,11 +2,11 @@
...
+<img src="<?= $this->img['src'] ?>"<?php if ($this->img['srcset'] !== $this->img['src']): ?> srcset="<?= $this->img['srcset'] ?>"<?php endif; ?><?php if (!empty($this->img['sizes'])): ?> sizes="<?= $this->img['sizes'] ?>"<?php endif; ?><?php if (!empty($this->img['width']) && !empty($this->img['height'])): ?> width="<?= $this->img['width'] ?>" height="<?= $this->img['height'] ?>"<?php endif; ?> alt="<?= $this->alt ?>"<?php if (!empty($this->img['loading'])): ?> loading="<?= $this->img['loading'] ?>"<?php endif; ?><?php if ($this->title): ?> title="<?= $this->title ?>"<?php endif; ?><?php if ($this->class || !empty($this->img['class'])): ?> class="<?= trim($this->class.' '.$this->img['class']) ?>"<?php endif; ?><?= $this->attributes ?> itemprop="image">
```
Comment (m-vo) [inline core-bundle/src/Resources/contao/templates/picture/picture_default.html5:9]: So images where `$this->img['hasSingleAspectRatio'])` is not set will now receive a value here, right? 

Doesn't this affect rendering for browsers that do not understand aspect ratios from `source` entries? :thinking:

Comment (ausi) [reply core-bundle/src/Resources/contao/templates/picture/picture_default.html5:9]: > So images where `$this->img['hasSingleAspectRatio'])` is not set will now receive a value here, right?

Correct.

> Doesn't this affect rendering for browsers that do not understand aspect ratios from `source` entries? 🤔

It shouldn’t matter as you size your images via CSS. We had a similar discussion in https://github.com/contao/contao/issues/937#issuecomment-568181620

**Comment by ausi** [comment]
Chrome 90 is released, so I think we can merge this. 🎉

**Review by fritzmg — APPROVED** [review]

**Review by Toflar — APPROVED** [review]

**Review by m-vo — CHANGES_REQUESTED** [review]
> This will also need to be adjusted in Image/Studio/_macros.html.twig for Contao 4.11.

Do we want to do this in this PR or in another one?

**Comment by ausi** [comment]
> Do we want to do this in this PR or in another one?

Another one I think because this one is for version 4.9.

**Comment by ausi** [comment]
PR for Contao 4.11: #2982

**Comment by leofeyer** [comment]
Thank you @ausi.

**Pull request merged**